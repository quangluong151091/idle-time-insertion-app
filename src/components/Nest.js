import React from 'react';
import {withRouter} from 'react-router-dom';
import { connect } from 'react-redux';
import * as actions from './timeline/actions';

class Nest extends React.Component  {
    componentDidMount() {
        this.props.fetchMachineStates(this.props.name);
    }
    handleClick = () => {
        this.props.history.push(`/${this.props.name}`)
    }
    render() {
        return (
            <tr onClick={this.handleClick.bind(this)}>
                <td className="imgNest"><img src={this.props.img} alt={this.props.name} /></td>
                <td className="nameNest"><h3>{this.props.name}</h3></td>
                <td className="noteNest"><span>{this.props.details}</span></td>
                <td className={this.props.status}></td>
            </tr>
        );
    }
}
function mapStateToProps(state) {
    return {
        status: state.fetchLatestStatus
    };
}
export default withRouter(connect(mapStateToProps, actions)(Nest))