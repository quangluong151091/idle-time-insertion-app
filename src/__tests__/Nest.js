import React from 'react';
import { shallow } from 'enzyme';

import Nest from '../components/Nest';

describe('Nest', () => {
  it('should render correctly', () => {

    const component = shallow(<Nest />);
  
    expect(component).toMatchSnapshot();
  });
});