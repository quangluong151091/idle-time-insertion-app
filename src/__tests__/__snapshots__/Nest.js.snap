// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Nest should render correctly 1`] = `
<Route>
  <Component />
</Route>
`;
